pkg/processor/builtin/util.go

```diff
@@ -12,28 +12,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package txfbuiltin
+package builtin
 
 import (
+	"context"
+	"github.com/conduitio/conduit/pkg/processor"
 	"strconv"
 	"time"
 
 	"github.com/conduitio/conduit/pkg/foundation/cerrors"
-	"github.com/conduitio/conduit/pkg/processor/transform"
 	"github.com/conduitio/conduit/pkg/record"
 )
 
+// funcProcessor is a stateless processor.Processor implementation
+// which is using a Go function to process records.
+// todo move out of this file.
+type funcProcessor struct {
+	fn func(context.Context, record.Record) (record.Record, error)
+}
```

Comment:
A nicer way to do this would be to use the same approach as [`http.HandlerFunc`](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/net/http/server.go;l=2080) ([godoc](https://pkg.go.dev/net/http#HandlerFunc)), and because `HandlerFunc` is so widely used I'd also rename it to `ProcessorFunc` to match the same naming.
```suggestion
type ProcessorFunc func(context.Context, record.Record) (record.Record, error)
```

Reply:
Hah, I was actually looking for a way to "wrap" all the built-in functions, and actually already had this. What I was missing is the `Execute` method, but for whatever reason I was convinced that I can't add any methods to `type Foo func()` :S

Reply:
How about we move this type to instance.go, where we have the Processor interface?

Reply:
Makes sense, do it!